src/Promact.TestCaseManagement.DomainModel/Models/TestCaseManagementBase.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace Promact.TestCaseManagement.DomainModel.Models
+{
+    public abstract class TestCaseManagementBase
+    {
+        [Key]
+        public int Id { get; set; }
+
+        public bool IsDeleted { get; set; }
```

Comment:
Same did not find any practical reason for making it nullable. Either project is deleted or not deleted at all. If i miss any point then please correct me.